tests/validation/cattlevalidationtest/core/test_ldap.py

```diff
@@ -0,0 +1,181 @@
+    assert user.profileUrl == user_copy.profileUrl
+    pass
+
+
+@if_ldap
+def test_ldap_search_get_group(admin_client):
+    searchGroup = os.getenv('LDAP_GROUP1', 'qualityAssurance')
+    searchGroupName = os.getenv('LDAP_GROUP_NAME', 'QA')
+    group = admin_client.list_identity(name=searchGroup)[0]
+    assert group.name == searchGroupName
+    group_copy = admin_client.by_id('identity', group.id)
+    assert group.name == group_copy.name
+    assert group.id == group_copy.id
+    assert group.login == group_copy.login
+    assert group.profilePicture == group_copy.profilePicture
+    assert group.profileUrl == group_copy.profileUrl
+
+
+def ldapClient(username, password):
+    token = requests.post(BASE_URL + 'token', {'code': username+':'+password})
+    token = token.json()
+    assert token['type'] != 'error'
+    token = token['jwt']
+    ldap_client = from_env(url=cattle_url())
+    ldap_client.valid()
+    ldap_client._auth = LdapAuth(token)
+    identities = ldap_client.list_identity()
+    assert len(identities) > 0
+    non_rancher = False
+    for identity in identities:
+        if (identity.externalIdType != 'rancher_id'):
+            non_rancher = True
+    assert non_rancher
+    return ldap_client
+
+
+@if_ldap
+def test_ldap_login(admin_client, cattle_url):
+    username = os.getenv('LDAP_USER1', 'sangeetha')
+    password = os.getenv('LDAP_USER1_password', 'Password1')
+    ldapClient(username, password)
+
+
+@if_ldap
+def test_ldap_incorrect_login():
+    username = os.getenv('LDAP_USER1', 'sangeetha')
+    token = requests.post(BASE_URL + 'token',
+                          {'code': username + ':' + random_str()})
+    token = token.json()
+    assert token['type'] == 'error'
+    assert token['status'] == 401
+    pass
+
+
+@if_ldap
+def test_ldap_unauthorized_login():
+    username = os.environ.get('API_AUTH_LDAP_'
+                              'SERVICE_ACCOUNT_PASSWORD',
+                              'Password1')
+    password = os.environ.get('API_AUTH_LDAP_'
+                              'SERVICE_ACCOUNT_USERNAME',
+                              'cattle')
+    token = requests.post(BASE_URL + 'token',
+                          {'code': username + ':' + password})
+    token = token.json()
+    assert token['type'] == 'error'
+    assert token['status'] == 401
+    pass
+
+
+@if_ldap
+def test_ldap_project_members():
+    username = os.getenv('LDAP_USER1', 'sangeetha')
+    password = os.getenv('LDAP_USER1_password', 'Password1')
+    user1_client = ldapClient(username, password)
+    username = os.getenv('LDAP_USER2', 'deniseschannon')
+    password = os.getenv('LDAP_USER2_password', 'Password1')
+    user2_client = ldapClient(username, password)
+
+    pass
+
+
+@if_ldap
+def test_ldap_project_create():
+    username = os.getenv('LDAP_USER1', 'sangeetha')
+    password = os.getenv('LDAP_USER1_password', 'Password1')
+    user1_client = ldapClient(username, password)
+    members = [{
```

Comment:
This seems odd. Can you explain what you're doing here with the odd member construct. Could you use a real user?